src/push/utils.ts

```diff
@@ -90,14 +90,51 @@ export function printBytes(bytes: number) {
   return `${bytes.toFixed(1)} ${BYTE_UNITS[exponent]}`;
 }
 
-export function getChunks<T>(arr: Array<T>, size: number): Array<T[]> {
+export function getChunks(arr: string[], size: number): string[][] {
   const chunks = [];
   for (let i = 0; i < arr.length; i += size) {
     chunks.push(arr.slice(i, i + size));
   }
   return chunks;
 }
 
+export function getSizedChunks<T extends { id?: string }>(
+  arr: Array<T>,
+  sizes: Map<string, number>,
+  maxChunkSizeKB: number,
+  maxChunkItems: number
+): Array<T[]> {
+  const chunks: Array<T[]> = [];
+  let currentChunk: T[] = [];
+  let currentSize = 0;
+
+  for (const item of arr) {
+    const sizeKb = item.id ? sizes.get(item.id) / 1024 : 1;
+
+    // If adding the current item would exceed limits, create a new chunk
+    if (
+      currentChunk.length >= maxChunkItems ||
+      currentSize + sizeKb > maxChunkSizeKB
+    ) {
+      chunks.push(currentChunk);
+      currentChunk = [];
+      currentSize = 0;
+    }
+    currentChunk.push(item);
+    currentSize += sizeKb;
+  }
+
+  // Push the last chunk if it contains any items
+  if (currentChunk.length > 0) {
+    chunks.push(currentChunk);
+  }
+
+  // log chunk sizes
+  progress(`Monitors will be pushed in ${chunks.length} chunks`);
```

Comment:
Lets try to move away from the `chunks` wording, user something like batches or requests seems appropriate.